Code/Tools/AssetProcessor/native/unittests/AssetRequestHandlerUnitTests.h

```diff
@@ -6,18 +6,34 @@
  *
  */
 #pragma once
+
 #if !defined(Q_MOC_RUN)
-#include "UnitTestRunner.h"
+#include <native/unittests/AssetProcessorUnitTests.h>
+#include <native/unittests/MockConnectionHandler.h>
 #endif
 
 namespace AssetProcessor
 {
+    class MockAssetRequestHandler;
+    class MockConnectionHandler;
+
     class AssetRequestHandlerUnitTests
-        : public UnitTestRun
+        : public QObject
+        , public UnitTest::AssetProcessorUnitTestBase
     {
         Q_OBJECT
-    public:
-        AssetRequestHandlerUnitTests();
-        virtual void StartTest() override;
+
+    protected:
+        void SetUp() override;
+        void TearDown() override;
+
+        AZStd::unique_ptr<MockAssetRequestHandler> m_requestHandler;
+
+        AssetProcessor::MockConnectionHandler m_connection;
```

Comment:
Should MockConnectionHandler m_connection be a AZStd::unique_ptr like m_requestHandler?